util/topgen.py

```diff
@@ -352,6 +354,36 @@ def generate_top_ral(top, ip_objs, out_path):
     gen_dv.gen_ral(top_block, str(out_path))
 
 
+def extract_copyright(*paths):
+    """Grep through any number of hjson files, extracting the lines that
+    hold copyright data. Duplicate data is pruned; the results will have
+    only unique lines of text."""
+
+    extract = list((re.compile(regex, re.IGNORECASE) for regex in (
+        r'.*(Copyright.*)|(.*\(c\).*)',
+        r'.*(SPDX-License-Identifier:.+)',
+        r'.*(Licensed under.+)',
+    )))
+    results = list((set() for _ in extract))
```

Comment:
Sets are unordered, but guarantee each item is unique. I relied upon this to scrape all the comments and avoid any duplicates.

In another comment you suggest using a license template. I used this approach because I was concerned different files would have different licenses, and I didn't want to omit any. However in practice the license in each file is identical.

If you're comfortable this will continue to be true in the future then it's easy enough to just use a template instead (maybe just read the first three lines of the file?)

Reply:
I've replaced this with a template as suggested.